tests/container_disk_test.go

```diff
@@ -174,22 +146,36 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 	Describe("[rfe_id:273][crit:medium][vendor:cnv-qe@redhat.com][level:component]Starting multiple VMIs", func() {
 		Context("with ephemeral registry disk", func() {
 			It("[test_id:1465]should success", func() {
-				num := 5
-				vmis := make([]*v1.VirtualMachineInstance, 0, num)
-				objs := make([]runtime.Object, 0, num)
-				for i := 0; i < num; i++ {
-					vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
-					// FIXME if we give too much ram, the vmis really boot and eat all our memory (cache?)
-					vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1M")
-					obj := tests.RunVMI(vmi, 10)
+				const count = 5
+				var vmis []*v1.VirtualMachineInstance
+				for i := 0; i < count; i++ {
+					vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Create(context.Background(), libvmi.NewCirros())
+					Expect(err).ToNot(HaveOccurred())
+					libwait.WaitForSuccessfulVMIStart(vmi)
 					vmis = append(vmis, vmi)
-					objs = append(objs, obj)
 				}
+				for _, vmi := range vmis {
+					pods, err := virtClient.CoreV1().Pods(testsuite.GetTestNamespace(vmi)).List(context.Background(), tests.UnfinishedVMIPodSelector(vmi))
+					Expect(err).ToNot(HaveOccurred())
 
-				for idx, vmi := range vmis {
-					verifyContainerDiskVMI(vmi, objs[idx])
+					By("Verifying the containerdisk is online")
+					containerdiskFound := false
```

Comment:
Can we avoid this nasty variables and just extract all the code below to a helper functions that returns this? e.g. `hasContainerDisk(...)`

Reply:
`hasContainerDisk` in a sense that it doesn't care how many?

Reply:
Thank you for the change. It is exactly what I meant.